extensions/amp-ad-network-cloudflare-impl/0.1/amp-ad-network-cloudflare-impl.js

```diff
@@ -0,0 +1,110 @@
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {AmpA4A} from '../../amp-a4a/0.1/amp-a4a';
+import {base64UrlDecodeToBytes} from '../../../src/utils/base64';
+import {dev} from '../../../src/log';
+
+/**
+ * Header that will contain Cloudflare generated signature
+ *
+ * @type {string}
+ * @private
+ */
+const AMP_SIGNATURE_HEADER = 'X-AmpAdSignature';
+
+/**
+ * Base URL of Ad server.  The extra /_a4a prefix signals Cloudflare
+ * A4A header processing to generate a signature and CORS headers.
+ *
+ * @type {string}
+ * @private
+ */
+//const CLOUDFLARE_BASE_URL = 'http://www.mycloudflaredomain.com/_a4a';
+
+/**
+ * This points to the internal "fake" cloudflare signed Ad server.
+ * Replace this with a real one defined above.
+ *
+ * @type {string}
+ * @private
+ */
+const CLOUDFLARE_BASE_URL =
+  '/extensions/amp-ad-network-cloudflare-impl/0.1/data';
```

Comment:
This follows the pattern from the fake impl to locate test data, so this represents the path to the test data folder.